### PR TITLE
pallet-vesting: Fix v1::post-migration when running try-runtime

### LINKED ISSUE #10275: pallet-vesting: v1::post-migration will panic when running try-runtime
Command:

```
RUST_LOG=runtime=trace,try-runtime::cli=trace,executor=trace 
./target/release/polkadot try-runtime
--execution Native --chain polkadot-dev
on-runtime-upgrade
live --uri ws://localhost:29944
```

Output:

```
2021-11-16 16:50:54.879 TRACE main runtime::staking: [0] 💸 inserting genesis staker: be5ddb1579b72e84524fc29e78609e3caf42e85aa118ebfe0b0ad404b5bdd25f (15Jbynf3...) => 1000000000000 => StakerStatus::Validator    
2021-11-16 16:50:54.880 TRACE main runtime::staking: [0] 💸 planning new session 0 at genesis    
2021-11-16 16:50:54.880 DEBUG main runtime::staking: [0] 💸 Starting the first era.    
2021-11-16 16:50:54.880  INFO main runtime::staking: [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2021-11-16 16:50:54.880 TRACE main runtime::staking: [0] 💸 planning new session 1 at genesis    
2021-11-16 16:50:54.880  WARN main runtime::parachains: Took active validators from set with wrong size    
2021-11-16 16:50:54.880  WARN main runtime::parachains: Took active validators from set with wrong size.    
2021-11-16 16:50:54.880  WARN main runtime::parachains: Took active validators from set with wrong size    
2021-11-16 16:50:54.880 TRACE main runtime::staking: [0] 💸 starting session 0    
2021-11-16 16:50:54.880  INFO main jsonrpsee_ws_client::transport: Connection established to target: Target { sockaddrs: [], host: "localhost", host_header: "localhost:29944", mode: Plain, path_and_query: "/" }    
2021-11-16 16:50:54.881  INFO main remote-ext: scraping key-pairs from remote @ 0x0dd2708f2a5ff138166dc27d297d9a05768c368afcf3fc2ab4e677538d86430a    
2021-11-16 16:50:54.881  INFO main remote-ext: downloading data for all pallets.    
2021-11-16 17:04:16.473  INFO main remote-ext: adding data for hashed key: 26aa394eea5630e07c48ae0c9558cef7f9cce9c888469bb1a0dceaa129672ef8    
2021-11-16 17:04:16.474  INFO main remote-ext: injecting a total of 1001421 keys    
2021-11-16 17:04:37.724 TRACE main executor: Request for native execution succeeded (native: polkadot-9130 (parity-polkadot-0.tx8.au0), chain: polkadot-9130 (parity-polkadot-0.tx8.au0))    
2021-11-16 17:04:37.725  INFO main jsonrpsee_ws_client::transport: Connection established to target: Target { sockaddrs: [], host: "localhost", host_header: "localhost:29944", mode: Plain, path_and_query: "/" }    
2021-11-16 17:04:37.725  INFO main try-runtime::cli: found matching spec name: "polkadot"    
2021-11-16 17:04:37.725  WARN main try-runtime::cli: spec version mismatch (local 9130 != remote 9122). This could cause some issues.    
2021-11-16 17:04:37.726 TRACE main executor: Request for native execution succeeded (native: polkadot-9130 (parity-polkadot-0.tx8.au0), chain: polkadot-9130 (parity-polkadot-0.tx8.au0))    
2021-11-16 17:04:37.726  INFO main polkadot_runtime: try-runtime::on_runtime_upgrade polkadot.    
2021-11-16 17:04:37.726  INFO main runtime::staking: [7725465] 💸 👜 staking bags-list migration passes PRE migrate checks ✅    
2021-11-16 17:04:37.726  INFO main runtime::staking: [7725465] 💸 migrating staking to Releases::V8_0_0    
2021-11-16 17:04:37.726 DEBUG main runtime::bags_list: [7725465] 👜inserted 28ee403d79d6fb7a1d3eb608ba1655ae12913e478176167307ee5bf81310e485 (1vfh4Wtf...) with weight 6298017085253 into bag 6923202753807, new count is 1    
2021-11-16 17:04:37.726 DEBUG main runtime::bags_list: [7725465] 👜inserted 0facb5c3d191e7e1fcbf8888bf0fb988dd1ae17668eb4025aba0855ec8d11f23 (1MZ2PGw1...) with weight 5849874653733 into bag 6219344874498, new count is 2    
...
2021-11-16 17:04:38.226 DEBUG main runtime::bags_list: [7725465] 👜inserted 1b2d25fb29f7e1424e938cfad6bf0b644430676955c055d18b33f7f589617936 (1cdhsnff...) with weight 1704107319053 into bag 1717891425287, new count is 17528    
2021-11-16 17:04:38.226  INFO main runtime::staking: [7725465] 💸 👜 completed staking migration to Releases::V8_0_0 with 17528 voters migrated    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ⚠️ System declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `StorageVersion(0)`    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Crowdloan    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Auctions    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Slots    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Registrar    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ParaSessionInfo    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Hrmp    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Ump    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Dmp    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Initializer    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Paras    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ParaScheduler    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ParaInherent    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ParaInclusion    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ParasShared    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ⚠️ Configuration declares internal migrations (which *might* execute). On-chain `StorageVersion(1)` vs current storage version `StorageVersion(1)`    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ParachainsOrigin    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for BagsList    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ElectionProviderMultiPhase    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Tips    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Bounties    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Multisig    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Proxy    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Identity    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Utility    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ⚠️ Vesting declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `StorageVersion(0)`    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Claims    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Treasury    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for TechnicalMembership    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for PhragmenElection    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for TechnicalCommittee    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Council    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Democracy    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for AuthorityDiscovery    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for ImOnline    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Grandpa    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Session    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Historical    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ⚠️ Offences declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `StorageVersion(0)`    
2021-11-16 17:04:38.226  INFO main runtime::offences: have 0 deferred offences, applying.    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ⚠️ Staking declares internal migrations (which *might* execute). On-chain `StorageVersion(0)` vs current storage version `StorageVersion(0)`    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Authorship    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for TransactionPayment    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Balances    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Indices    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Timestamp    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Babe    
2021-11-16 17:04:38.226  INFO main runtime::frame-support: ✅ no migration for Scheduler    
2021-11-16 17:04:38.316 DEBUG main runtime::bags_list: [7725465] 👜count of nodes: 17528    
2021-11-16 17:04:38.352 DEBUG main runtime::bags_list: [7725465] 👜count of active bags 117    
2021-11-16 17:04:42.373  INFO main runtime::staking: [7725465] 💸 👜 staking bags-list migration passes POST migrate checks ✅    

====================

Version: 0.9.13-a711993f93-x86_64-linux-gnu

   0: sp_panic_handler::set::{{closure}}
   1: std::panicking::rust_panic_with_hook
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:628:17
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:519:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/sys_common/backtrace.rs:141:18
   4: rust_begin_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:517:5
   5: core::panicking::panic_fmt
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/panicking.rs:101:14
   6: pallet_vesting::migrations::v1::post_migrate
   7: <(TupleElement0,TupleElement1) as frame_support::traits::hooks::OnRuntimeUpgrade>::post_upgrade
   8: frame_executive::Executive<System,Block,Context,UnsignedValidator,AllPallets,COnRuntimeUpgrade>::try_runtime_upgrade
   9: polkadot_runtime::api::dispatch
  10: <polkadot_client::PolkadotExecutorDispatch as sc_executor::native_executor::NativeExecutionDispatch>::dispatch
  11: std::thread::local::LocalKey<T>::with
  12: sc_executor::native_executor::WasmExecutor::with_instance::{{closure}}
  13: sc_executor::wasm_runtime::RuntimeCache::with_instance
  14: <sc_executor::native_executor::NativeElseWasmExecutor<D> as sp_core::traits::CodeExecutor>::call
  15: sp_state_machine::execution::StateMachine<B,H,Exec>::execute_aux
  16: sp_state_machine::execution::StateMachine<B,H,Exec>::execute
  17: try_runtime_cli::state_machine_call
  18: try_runtime_cli::TryRuntimeCmd::run::{{closure}}
  19: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
  20: sc_cli::runner::main::{{closure}}::{{closure}}::{{closure}}
  21: <futures_util::future::poll_fn::PollFn<F> as core::future::future::Future>::poll
  22: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  23: tokio::park::thread::CachedParkThread::block_on
  24: tokio::runtime::Runtime::block_on
  25: sc_cli::runner::Runner<C>::async_run
  26: polkadot_cli::command::run
  27: polkadot::main
  28: std::sys_common::backtrace::__rust_begin_short_backtrace
  29: std::rt::lang_start::{{closure}}
  30: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/core/src/ops/function.rs:259:13
      std::panicking::try::do_call
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panic.rs:129:14
      std::rt::lang_start_internal::{{closure}}
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/rt.rs:45:48
      std::panicking::try::do_call
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/panic.rs:129:14
      std::rt::lang_start_internal
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/std/src/rt.rs:45:20
  31: main
  32: __libc_start_main
  33: _start


Thread 'main' panicked at 'A bounded vec with incorrect count of items was created.', /home/koushiro/.cargo/git/checkouts/substrate-7e08433d4c370a21/c087bbe/frame/vesting/src/migrations.rs:72

This is a bug. Please report it at:

	https://github.com/paritytech/polkadot/issues/new

2021-11-16 17:04:42.390  WARN main wasm-runtime: Evicting failed runtime instance: RuntimePanicked("A bounded vec with incorrect count of items was created.")    
Error: 
   0: Invalid input: failed to execute 'TryRuntime_on_runtime_upgrade': RuntimePanicked("A bounded vec with incorrect count of items was created.")

Backtrace omitted.
Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

Fix #10275

### CODE REVIEW

**Comment by bkchr** [comment]
Already done https://github.com/paritytech/substrate/pull/10141

**Comment by koushiro** [comment]
> Already done #10141

it's different :rofl:

**Comment by bkchr** [comment]
I see, but your code is also wrong.

File: frame/vesting/src/lib.rs [hunk 1]
```diff
@@ -199,7 +199,11 @@ pub mod pallet {
 
 		#[cfg(feature = "try-runtime")]
 		fn post_upgrade() -> Result<(), &'static str> {
-			migrations::v1::post_migrate::<T>()
+			if StorageVersion::<T>::get() == Releases::V0 {
+				migrations::v1::post_migrate::<T>()
+			} else {
+				Ok(())
+			}
```
Comment (bkchr) [inline frame/vesting/src/lib.rs:206]: This doesn't make sense. Because the `StorageVersion` should be `V1` after applying the migration.

Comment (bkchr) [reply frame/vesting/src/lib.rs:206]: Where is it failing actually?

Comment (koushiro) [reply frame/vesting/src/lib.rs:206]: https://github.com/paritytech/substrate/blob/master/frame/vesting/src/migrations.rs#L72

Comment (bkchr) [reply frame/vesting/src/lib.rs:206]: And when is this failing? Do you already have applied this upgrade?

Comment (koushiro) [reply frame/vesting/src/lib.rs:206]: I just tried the command beblow

```
RUST_LOG=runtime=trace,try-runtime::cli=trace,executor=trace 
./target/release/polkadot try-runtime
--execution Native --chain polkadot-dev
on-runtime-upgrade
live --uri ws://localhost:29944 --at 0x83c1812face22aa9d31f6d0033874a88bf5731d88a9ab8ed44cc727a165ea20b
```

**Comment by koushiro** [comment]
I'm sorry, I seem to accidentally submit to the upstream during the rebase process. I closed this PR and will describe the error message in the #10275

**Pull request closed without merging**